components/charts/SpendingTrendsChart.tsx

```
@@ -0,0 +1,223 @@
+      if (trendType === 'daily' || trendType === 'last30days') {
+        return period?.toString() || '';
+      } else {
+        // For weekly view, show only every other label to reduce crowding
+        // Find the index of this period in our chart data
+        const dataIndex = chartData.findIndex(item => item.period === period);
+        if (dataIndex !== -1 && dataIndex % 2 !== 0) {
+          return '';
+        }
+        const str = period?.toString() || '';
+        return str.length > 6 ? str.substring(0, 6) : str;
+      }
```

Biome (1.9.4)

[error] This else clause can be omitted because previous branches break early.

Unsafe fix: Omit the else clause.


(lint/style/noUselessElse)